File: backend/app/schemas/super_admin.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class TenantOverview(BaseModel):
    id: str
    name: str
    slug: str
    is_active: bool
    user_count: int
    created_at: datetime
    subscription_status: str | None = None
    plan_name: str | None = None
    last_sync_at: datetime | None = None
    sync_status: str | None = None  # green, yellow, red


class SystemHealth(BaseModel):
    total_tenants: int
    active_tenants: int
    inactive_tenants: int
    total_users: int
    active_users: int
    total_jobs_24h: int
    failed_jobs_24h: int
    redis_connected: bool
    db_connected: bool


class SuperDashboard(BaseModel):
    system_health: SystemHealth
    tenants: list[TenantOverview]
    billing_mrr: Decimal
    billing_active: int
    billing_past_due: int

File: frontend/src/types/super-admin.ts
export interface TenantOverview {
  id: string;
  name: string;
  slug: string;
  is_active: boolean;
  user_count: number;
  created_at: string;
  subscription_status: string | null;
  plan_name: string | null;
  last_sync_at: string | null;
  sync_status: "green" | "yellow" | "red" | null;
}

export interface SystemHealth {
  total_tenants: number;
  active_tenants: number;
  inactive_tenants: number;
  total_users: number;
  active_users: number;
  total_jobs_24h: number;
  failed_jobs_24h: number;
  redis_connected: boolean;
  db_connected: boolean;
}

export interface SuperDashboard {
  system_health: SystemHealth;
  tenants: TenantOverview[];
  billing_mrr: string;
  billing_active: number;
  billing_past_due: number;
}
